### DISCUSSION TITLE
Get quickly structure and admincode (digits/letters) of administrative areas of a country

### DISCUSSION THREAD

**Discussion by stouch** [osm-search/Nominatim#3111] [Show and tell]
I needed to get the structure of OSM regions/states/cantons of  some countries, and their sub aeras.

For each region and sub-regions, I needed to get their adminCode (in France, we have 2-digits codes for each *départements*, but sometimes for some other countries it's 3-letters identifiers, etc.).

Anyway, just wanted to show you this : https://github.com/stouch/osm-to-admincodes-mapping

It uses ChatGPT and your Nominatim full-planet instance.

Regards,

**Comment by stouch** [comment]
Just updated the repo and introduced some other mappings

**Comment by lonvia** [comment]
Please be aware that systematic downloads like this are against the [usage policy](https://operations.osmfoundation.org/policies/nominatim/) of nominatim.openstreetmap.org. Have a look at Overpass API or https://osm-boundaries.com/.

**Reply by stouch** [reply]
Hi, I run this generation on my self-hosted instance. I should prevent this adding a warning in the .env file

**Reply by stouch** [reply]
Thanks @lonvia for osm-boundaries, it's interesting to get quickly the countries structures from openstreetmap data.

**Reply by stouch** [reply]
@lonvia  I added a disclaimer about Policy and removed the public API URL from .env.sample
